tests/playbooks/integration_pytest_python3.yml

```diff
@@ -42,38 +42,66 @@
         path: "{{ rundir }}"
         recurse: true
 
-    - command: git rev-parse --show-toplevel
-      register: git_top_directory
-      delegate_to: localhost
-
-    - debug:
-        var: git_top_directory
-
-    - synchronize:
-        src: "{{ git_top_directory.stdout }}/"
-        dest: "{{ rundir }}/"
-        recursive: yes
-        delete: yes
-        rsync_opts:
-          - "--exclude=.pyc"
-          - "--exclude=__pycache__"
-      when: False
-
-    # TODO: using tar and copying the file is a workaround for the synchronize
-    # module that does not work in test-harness. Related issue:
-    # https://github.com/linux-system-roles/test-harness/issues/102
-    #
-    - name: Create Tar file
-      shell: 'tar -cvf {{ git_top_directory.stdout }}/testrepo.tar
-             --exclude "*.pyc" --exclude "__pycache__" --exclude testrepo.tar
-             -C {{ git_top_directory.stdout }} .'
-      delegate_to: localhost
+    - name: Determine legacy role format or collection format
```

Comment:
Might I suggest something like this instead?  https://gist.github.com/richm/f3338790eb48b6a3f0963f908a2f4a03

That is
* assume that if running as a role, the tests directory will be `../../tests`, the modules directory will be `../../library`, and the module_utils directory will be `../../module_utils`
* otherwise, look in the various ansible paths, including the collection paths, for the modules and module_utils

Reply:
Hi @richm, could you please review this commit?
https://github.com/linux-system-roles/network/pull/319/commits/ce7b70dd2f93cc3283317149caef7892d9e0d6df

I had to tweak somehow to make it work on the control node. Hope it's ok...

Good news is staging tests with ansible 2.9 successfully run now!